include/llama/Tuple.hpp

```diff
@@ -27,58 +26,96 @@ namespace llama
 
         constexpr Tuple() = default;
 
-        LLAMA_FN_HOST_ACC_INLINE
-        constexpr Tuple(FirstElement first, Elements... rest) : first(first), rest(rest...)
+        template <
+            typename T,
+            typename... Ts,
+            typename
+            = std::enable_if_t<std::is_constructible_v<FirstElement, T> && std::is_constructible_v<RestTuple, Ts...>>>
+        LLAMA_FN_HOST_ACC_INLINE constexpr Tuple(T&& firstArg, Ts&&... restArgs)
+            : first(std::forward<T>(firstArg))
+            , rest(std::forward<Ts>(restArgs)...)
         {
         }
 
         LLAMA_FN_HOST_ACC_INLINE
-        constexpr Tuple(FirstElement first, Tuple<Elements...> rest) : first(first), rest(rest)
+        constexpr Tuple(FirstElement first, Tuple<Elements...> rest) : first(std::move(first)), rest(std::move(rest))
         {
         }
 
         FirstElement first; ///< the first element (if existing)
-        RestTuple rest; ///< the remaining elements
-    };
-
-    template <typename T_FirstElement>
-    struct Tuple<T_FirstElement>
-    {
-        using FirstElement = T_FirstElement;
-        using RestTuple = Tuple<>;
-
-        constexpr Tuple() = default;
-
-        LLAMA_FN_HOST_ACC_INLINE
-        constexpr Tuple(FirstElement const first, Tuple<> const rest = Tuple<>()) : first(first)
-        {
-        }
-
-        FirstElement first;
+        [[no_unique_address]] RestTuple rest; ///< the remaining elements
```

Comment:
Did LLAMA switch to C++20? The description still says its C++17.

Reply:
LLAMA compiles and works fine with C++17. However, there are several additional features available with C++20, hidden behind `#ifdef` etc. Unknown attributes in particular, like this one in C++17 mode, will just be ignored by the compiler, so we are fine.